protocols/request-response/src/codec.rs

```diff
@@ -76,3 +77,35 @@ pub trait Codec {
     where
         T: AsyncWrite + Unpin + Send;
 }
+
+const DEFAULT_READ_STEP_SIZE: u64 = 1024;
+
+/// Reads a message from the given socket.
+///
+/// The `max_size` parameter is the maximum size in bytes of the message that we accept. This is
+/// necessary in order to avoid DoS attacks where the remote sends us a message of several
+/// gigabytes.
+pub(crate) async fn read_to_end(
+    socket: &mut (impl AsyncRead + Unpin),
+    max_size: usize,
+) -> io::Result<Vec<u8>> {
+    let step_size = min(DEFAULT_READ_STEP_SIZE, u64::try_from(max_size).unwrap());
+    let mut res = Vec::with_capacity(usize::try_from(step_size).unwrap());
```

Comment:
If we declare `DEFAULT_READ_STEP_SIZE` as `usize`, can we avoid all the conversions?